android/app/src/main/java/com/zulipmobile/ShareFileAndroid.kt

```diff
@@ -0,0 +1,41 @@
+package com.zulipmobile
+
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import android.content.Intent
+import android.net.Uri
+import com.facebook.react.bridge.Promise
+import com.imagepicker.FileProvider
+import java.io.File
+
+/**
+ * Allows Sending Files from Zulip to other apps.
+ */
+class ShareFileAndroid(reactContext: ReactApplicationContext?) : ReactContextBaseJavaModule(reactContext) {
+    override fun getName(): String {
+        return "ShareFileAndroid"
+    }
+
+    @ReactMethod
+    fun shareFile(path: String, promise: Promise) = try {
+        val sharedFile = File(path)
+        val sharedFileUri: Uri = FileProvider.getUriForFile(
+            reactApplicationContext,
+            reactApplicationContext.packageName + ".provider",
+            sharedFile)
+
+        val shareIntent = Intent()
+        shareIntent.action = Intent.ACTION_SEND
+        shareIntent.type = reactApplicationContext.contentResolver.getType(sharedFileUri)
+        shareIntent.putExtra(Intent.EXTRA_STREAM, sharedFileUri)
+
+        val chooserIntent = Intent.createChooser(shareIntent,
+            reactApplicationContext.resources.getText(R.string.send_to))
+        chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        reactApplicationContext.startActivity(chooserIntent)
+        promise.resolve("success")
```

Comment:
I changed it to `promise.resolve(null)` in the latest revision, at JS side it resolves to null, which would make the type of function - `Promise<null>`. I couldn't find a way to resolve `undefined` explicitly from Kotlin since `Promise.resolve` always requires a parameter, if there is a way to do this let me know and I will make the necessary changes.

Reply:
Sounds good, thanks! `Promise<null>` should be clear enough.